### PR TITLE
CCL terms of reference

### PR DESCRIPTION
**Pull request by elisachisari**
Adds the terms of reference discussed in the telecons and the Slack channel.

### CODE REVIEW

File: TERMS.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+CCL interface support is as follows.
+
+- Public APIs are maintained in Python only (i.e., version bumping is not necessary following changes in the C interface).
+
+- The C implementation can be used for speed as needed.
+
+- Documentation should be maintained in the main README, readthedocs and the benchmarks folder.
+
+- Examples will be maintained in a separate repository.
```
Comment (philbull) [inline TERMS.md]: Just double-checking -- does this mean we will create a new `ccl-examples` repo or similar?

File: TERMS.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+Boundaries of CCL (with respect to various models and systematics)
+
+- Not all models need to have equal support. CCL aims to support the DESC analyses, not to support all possible statistics in all models.
+
+- CCL should provide a generic framework to enable modeling of systematics effects. However, overly specific models for systematics are best located in other repositories (e.g., firecrown).
```
Comment (philbull) [inline TERMS.md]: Might be worth giving an example of what would count as _overly_ specific.

File: TERMS.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+- Contributions to CCL should be well-documented in python (only doc-string level docs are required, readthedocs does the rest). C code is expected to be commented, though not documented.
+
+- Features added to CCL should pass required tests.
+
+	1. one (publicly available) benchmark with an independent prediction should be provided to test the specific feature that has been added. The benchmark code can use CCL input as long as that is not affecting the validation procedure. 
+	   
+	2. Tests should run in continuous integration scheme used for CCL.
+
+	3. Unit tests should be written as identified in code review.
```
Comment (philbull) [inline TERMS.md]: This point is a little vague for people who aren't familiar with the code review.

File: TERMS.md (hunk 1 above)
Comment (beckermr) [reply TERMS.md]: Yes

**Pull request merged**